File: UniversityApiBackend/DTOs/Courses/CourseCreateDTO.cs
using System.ComponentModel.DataAnnotations;
using UniversityApiBackend.DTOs.Categories;
using UniversityApiBackend.DTOs.Chapters;

using UniversityApiBackend.Models.DataModels;

namespace UniversityApiBackend.DTOs.Courses
{
    public class CourseCreateDTO
    {
        public int Id { get; set; }
        [Required, StringLength(50)]
        public string Name { get; set; } = string.Empty;

        [Required, StringLength(280)]
        public string ShortDescription { get; set; } = string.Empty;
        [Required]
        public string Description { get; set; } = string.Empty;

        [Required]
        public string PublicGoal { get; set; } = string.Empty;

        public Level Levels { get; set; } = Level.Basic;

public List<CategoryCourseDTO> Categories { get; set; }             


    }
}

File: Frontend/front-university/src/app/services/api/models/course-create-dto.ts
/* tslint:disable */
/* eslint-disable */
import { CategoryCourseDto } from './category-course-dto';
import { ChapterDto } from './chapter-dto';
import { Level } from './level';
export interface CourseCreateDto {
  categories?: null | Array<CategoryCourseDto>;
  description: string;
  id?: number;
  levels?: Level;
  name: string;
  publicGoal: string;
  shortDescription: string;
  chapter:ChapterDto;

}
